TestClass.cpp

```diff
@@ -0,0 +1,234 @@
+#include <iostream>
+#include <cmath>
+#include <cassert>  
+
+using namespace std; //use std::cout
+
+#define FILE (__FILE__)
+#define LINE (__LINE__)
+
+// #define LOG_LENGTH (300)
+const int LOG_LENGTH = 300;
+
+void print_log(const char *filename,
+               int         line,
+               const char *message)
+{
+    printf("F:%s L:%d : %s\n", filename, line, message);
+}
+
+class TestClass
+{
+    private:
+        int var1;
+        int var2;
+        // double var3;
+        // double var4;
+        int max_average_value;
+        int max_square_root_value;
+
+    public:
+    //get and set functions are public
+
+        TestClass(int a, int b)
+        {
+            var1 = a;
+            var2 = b;
+            max_average_value = 10;
+            max_square_root_value = 7;
+        }
+
+#if 0
+        // do not create constructors with same number of parameters, ideally different number of parameters
+        TestClass(double a, double b)
+        {
+            var3 = a;
+            var4 = b;
+        }
+
+        double getSum(double var1, double var2)
+        {
+            return (var3 + var4);
+        }
+#endif        
+
+        int getSum(void)
+        {
+            return (var1 + var2);
+        }
+
+        int getAverage(void)
+        {
+            return ((var1 + var2)/2);
+        }
+
+        int getMaxAverageValue(void)
+        {
+            return max_average_value;
+        }
+
+        bool isAverageWithinDesiredMaxValue(void) //remove void
+        {
+            if (getAverage() > getMaxAverageValue())
```

Comment:
use assertions for the code instead of if-else. Check the example I gave initially
It uses a macro that looks nicer (assert behavior is similar and also good)

```
#define IS_TRUE(x) { if (!(x)) std::cout << __FUNCTION__ << " failed on line " << __LINE__ << std::endl; }

// Test for function1()
// You would need to write these even when using a framework
void test_function1()
{
    IS_TRUE(!function1(0));
    IS_TRUE(!function1(5));
    IS_TRUE(function1(10));
}
```

Reply:
Thank you for explaining the test functions.. makes sense. Does that mean it's bad practice to perform input parameter checks WITHIN the code? Are input param checks supposed to be only as part of the UT cases?